pages/behaviors/manage.tsx

```diff
@@ -33,28 +32,53 @@ interface BehaviorAsProps {
     _id: string;
 }
 
-export default function manageBehaviorsPage({ behaviors }) {
+export default function manageBehaviorsPage() {
     // Verifies if user has the correct permissions
     const {allowed, role} = CheckUser(["Admin", "BCBA"])
     if(!allowed) return(<div>Redirecting...</div>);
 
-    const [behaviorList, setBehaviorList] = useState<BehaviorAsProps[]>(
-        behaviors.map((behavior: Behavior, idx: number) => ({
-            behaviorName: behavior.name,
-            description: behavior.description,
-            datatype: behavior.datatype,
-            id: idx + 1,
-            _id: behavior._id,
-        }))
-    );
+    // fetch the behavior data on the client side
+    const [behaviors, setBehaviors] = useState(null)
+    const [behaviorList, setBehaviorList] = useState([]);

```

Comment:
I would be more specific here with the type of your state. Please refer to my comment above for more details.

Reply:
It would be better to use Behavior from prisma client than the behavior interface defined here for annotating type to "behaviors", correct?

Reply:
Yeah I think so. As far as I know, it's usually better to have a single source of truth for our models. 
Try using the Behavior type from the prisma client, and if you run into issues with that let me know.